src/Illuminate/Log/LogManager.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace Illuminate\Log;
+
+use Psr\Log\AbstractLogger;
+use Psr\Log\LoggerInterface;
+use Illuminate\Contracts\Events\Dispatcher;
+
+class LogManager extends AbstractLogger implements LoggerInterface
+{
+    /**
+     * The available log channels.
+     *
+     * @var \Illuminate\Log\Channel[]
+     */
+    protected $channels = [];
+
+    /**
+     * The default log channels.
+     *
+     * @var string[]
+     */
+    protected $defaultChannels = [];
+
+    /**
+     * The events dispatcher.
+     *
+     * @var \Illuminate\Contracts\Events\Dispatcher
+     */
+    protected $events;
+
+    /**
+     * Create the LogManager.
+     *
+     * @param  \Illuminate\Contracts\Events\Dispatcher  $events
+     */
+    public function __construct(Dispatcher $events)
+    {
+        $this->events = $events;
+    }
+
+    /**
+     * Retrieve a channel to use for logging.
+     *
+     * @param  string  $channel
+     * @return \Illuminate\Log\Channel
```

Comment:
Surely this should be the channel contract?

Reply:
Definitely, thanks! Updated

Reply:
Updated all references to the concrete